internal/certs/certs.go

```diff
@@ -0,0 +1,153 @@
+package certs
+
+import (
+	"crypto/x509/pkix"
+	"fmt"
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/pkg/errors"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"path/filepath"
+	"time"
+)
+
+type CertManager struct {
+	CertificatesDir string
```

Comment:
Can we outsource file handling to some external entity? Have this struct focus on the logical, pure side of things - create the keys and certs, create the kubeconfig, but not inside files. We're gonna need it when we move away from using files for everything. Also this is sensitive info, it should ideally avoid the disk as much as possible

Reply:
This file contains a single line that actually writes certificates to a file.
We can change it once we no longer pass the certs as files.